docs/cli/function/layers.md

```diff
@@ -0,0 +1,219 @@
+
+Any node module that is in the layer's `node_modules` folder can be accessed from the function as if the node module is in the function's `node_modules` folder.
+
+*In order to take advantage of Lambda layer's for your NodeJS function, you don't even need to update your function's code!*
+
+</amplify-block>
+
+<amplify-block name="Python">
+
+A `python` folder is auto-generated for you. In there you'll find a skeleton folder structure to add your Python packages to. Add your packages to `lib/python3.8/site-packages` folder and then you can use it from within your Python function same as any other `import`. 
+
+</amplify-block>
+
+</amplify-block-switcher>
+
+### Add shared assets
+
+Any assets like large images or other files that you want to share across various functions can be placed in the `amplify/backend/function/<layer-name>/opt/` folder. Your function's code can import any assets by looking for files in the `opt/` path.
+
+## Add a layer to a function
+
+You can either create a new function and add Lambda layers by running `amplify add function` or add layers to an existing function using `amplify update function`. Select `Lambda function` when prompted and you'll be presented the following question during the guided flow: 
+
+```console
+? Do you want to configure Lambda layers for this function? Yes
+? Provide existing layers or select layers in this project to access from this function (pick up to 5): (Press <space> to select, <a> to toggle all, <i> to invert selection)
+❯◯ Provide existing Lambda Layer ARNs
+ ◯ myamplifylayer1
+ ◯ myamplifylayer2
+```
+
+You can either add an existing layer in AWS by referencing its ARN or select a layer from your Amplify project that's listed below.
+
+When adding a layer from your Amplify project, you'll also be able to select the layer version. Amplify CLI automatically versions your Lambda layers by recognizing changes to the layer's folder contents. The largest layer version number represents the most recent changes. 
+```console
+? Select a version for myamplifylayer1:
+*> 8*
+  5
+  3
+  2
+  1
+```
+
+Given that layers can have overlapping contents, the order of the layer matters. You can adjust the layer's order if needed in the next step.
+
+```console
+? Modify the layer order:
+(Layers with conflicting files will overwrite contents of layers earlier in the list):
+- layer2
+- layer3
+- layer6
+- <ARN1>
+- <ARN2>
+```
+
+Now, you've successfully added a layer to your function.
+
+## Deploy Lambda layers & functions with Lambda layers
+
+Once you're ready with your changes in your layer and functions, you can deploy them by running `amplify push`. 
+
+If a Layer’s content has been updated and it has permissions associated, Amplify CLI will prompt you whether you want to carry the permissions forward to a newer version.
+
+```console
+Content changes in Lambda layer {LAYER_NAME} detected. What permissions do you want to grant to this new layer version?
```

Comment:
We need a standard way of representing a layer name.

"\<lambda-layer-name\>", `layer-name`, {LAYER_NAME}